src/java/arjdbc/jdbc/RubyJdbcConnection.java

```diff
@@ -3627,7 +3656,10 @@ protected final TableName extractTableName(
     }
 
     protected IRubyObject valueForDatabase(final ThreadContext context, final IRubyObject attribute) {
-        return attribute.callMethod(context, "value_for_database");
+        if (attribute.respondsTo("value_for_database")) {
```

Comment:
would like to understand, why is this necessary - what would not respond_to? `value_for_database`?
asking since it's now double the method lookup cost and I wonder if this is a hot spot.

Reply:
gone with the latest version